homeassistant/components/mqtt/select.py

```diff
@@ -117,23 +125,27 @@ class MqttSelect(MqttEntity, SelectEntity, RestoreEntity):
 
     _attributes_extra_blocked = MQTT_SELECT_ATTRIBUTES_BLOCKED
 
-    def __init__(self, hass, config, config_entry, discovery_data):
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        config: ConfigType,
+        config_entry: ConfigEntry,
+        discovery_data: DiscoveryInfoType | None,
+    ) -> None:
         """Initialize the MQTT select."""
-        self._config = config
-        self._optimistic = False
-        self._sub_state = None
-
-        self._attr_current_option = None
+        self._templates: dict[str, Callable[..., Any]]
```

Comment:
This is just a type hint. It should be a class attribute.

Reply:
You mean It should be out of `__init__()`?. A now moved to direct at class level.

Reply:
I reworked the class attributes, moved all assigns to `_setup_from_config`. Since this is called when a discovery update is processed.